Symfony/CS/Tests/Fixer/Contrib/DirConstantFixerTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Tests\Fixer\Contrib;
+
+use Symfony\CS\Tests\Fixer\AbstractFixerTestBase;
+
+/**
+ * @author Vladimir Reznichenko <kalessil@gmail.com>
+ */
+class DirConstantFixerTest extends AbstractFixerTestBase
+{
+    /**
+     * @dataProvider provideExamples
+     */
+    public function testFix($expected, $input = null)
+    {
+        $this->makeTest($expected, $input);
+    }
+
+    public function provideExamples()
+    {
+        return array(
+            array('<?php $x = "dirname";'),
+
+            array('<?php $x = dirname(__FILE__.".dist");'),
+
+            array('<?php $x = Class::dirname(__FILE__);'),
+            array('<?php $x = Scope\dirname(__FILE__);'),
+            array('<?php $x = $object->dirname(__FILE__);'),
+
+            array('<?php $x = new \dirname(__FILE__);'),
+            array('<?php $x = new dirname(__FILE__);'),
+            array('<?php $x = new Scope\dirname(__FILE__);'),
+
+            array('<?php $x = __DIR__;', '<?php $x = dirname(__FILE__);'),
```

Comment:
Probably - at the end first support line will be CS-Fixer project, so contrib-level would fit as for me.